### PR TITLE
Added TensorFlow sample

### PR DESCRIPTION
**Pull request by JuliusKunze**
This example might be interesting for people familiar with machine learning. Feel free to close this Pull Request if this is too specific.

Tested on OSX.

### CODE REVIEW

File: samples/tensorflow/HelloTensorflow.kt [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    data[0] = value
+
+    fun intTensorDeallocator(data: COpaquePointer?, len: size_t, arg: COpaquePointer?) {
+        nativeHeap.free(data!!.reinterpret<IntVar>())
+    }
+
+    return TF_NewTensor(TF_INT32,
+            dims = null, num_dims = 0,
+            data = data, len = IntVar.size,
+            deallocator = staticCFunction(::intTensorDeallocator), deallocator_arg = null)!!
```
Comment (olonho) [inline samples/tensorflow/HelloTensorflow.kt]: Note that you may use lambda for deallocator here.

File: samples/tensorflow/HelloTensorflow.kt [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+typealias Status = CPointer<TF_Status>
+typealias Operation = CPointer<TF_Operation>
+typealias Tensor = CPointer<TF_Tensor>
+
+val Status.isOk: Boolean get() = TF_GetCode(this) == TF_OK
+val Status.errorMessage: String get() = TF_Message(this)!!.toKString()
+fun Status.delete() = TF_DeleteStatus(this)
+fun Status.validate() {
+    if (!isOk) {
+        throw Error("Status is not ok: $errorMessage")
```
Comment (olonho) [inline samples/tensorflow/HelloTensorflow.kt]: Shalln't `delete` be called on error path as well?

File: samples/tensorflow/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# TensorFlow demo
+
+[Install TensorFlow for C](https://www.tensorflow.org/versions/r1.1/install/install_c) into `/usr/local`.
+
+Compile:
```
Comment (olonho) [inline samples/tensorflow/README.md]: Some description of what demo does would be good.

File: samples/tensorflow/tensorflow.def [hunk 4]
```diff
@@ -0,0 +1,6 @@
+headers = tensorflow/c/c_api.h
+headers.osx =
```
Comment (olonho) [inline samples/tensorflow/tensorflow.def]: Empty platform-specific headers seems not needed here.

**Comment by JuliusKunze** [comment]
@olonho Good points! I addressed all of them.

**Pull request merged**